src/utils/general.ts

```diff
@@ -122,76 +125,94 @@ function evaluate(
 
 const aggregateMap: {
   [key in QueryViewerAggregationType]: (
-    values: number[],
+    values: GxBigNumber[],
     quantities: number[]
-  ) => number;
+  ) => GxBigNumber;
 } = {
   [QueryViewerAggregationType.Sum]: (
-    values: number[],
+    values: GxBigNumber[],
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     _quantities: number[]
   ) => {
-    let sumValues: number = null;
+    let sumValues: GxBigNumber = null;
 
     for (let i = 0; i < values.length; i++) {
       if (values[i]) {
-        sumValues += values[i];
+        sumValues = sumValues ? add(sumValues, values[i]) : values[i];
       }
     }
+
     return sumValues;
   },
 
   [QueryViewerAggregationType.Average]: (
-    values: number[],
+    values: GxBigNumber[],
     quantities: number[]
   ) => {
-    let sumValues: number = null;
+    let sumValues: GxBigNumber = null;
     let sumQuantities: number = null;
 
     for (let i = 0; i < values.length; i++) {
-      if (values[i]) {
-        sumValues += values[i];
+      const value = values[i];
+      if (value) {
+        sumValues = sumValues ? add(sumValues, value) : value;
         sumQuantities += quantities[i];
       }
     }
-    return sumValues != null ? sumValues / sumQuantities : null;
+
+    return sumValues != null ? divide(sumValues, sumQuantities) : null;
   },
 
   [QueryViewerAggregationType.Count]: (
-    _values: number[],
+    _values: GxBigNumber[],
     quantities: number[]
-  ) => quantities.reduce((a, b) => a + b, 0),
+  ) => {
+    return new GxBigNumber(quantities.reduce((a, b) => a + b, 0));
+  },
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  [QueryViewerAggregationType.Max]: (values: number[], _quantities: number[]) =>
-    values.length === 0 ? null : Math.max(...values),
+  [QueryViewerAggregationType.Max]: (
+    values: GxBigNumber[],
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _quantities: number[]
+  ) =>
+    values.length === 0
+      ? null
+      : new GxBigNumber(Math.max(...values.map(Number))),
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  [QueryViewerAggregationType.Min]: (values: number[], _quantities: number[]) =>
-    values.length === 0 ? null : Math.min(...values)
+  [QueryViewerAggregationType.Min]: (
+    values: GxBigNumber[],
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _quantities: number[]
+  ) =>
+    values.length === 0
+      ? null
+      : new GxBigNumber(Math.min(...values.map(Number)))
 };
 
 export const aggregate = (
   aggregation: QueryViewerAggregationType,
-  values: number[],
+  values: GxBigNumber[],
   quantities: number[]
-) =>
-  aggregateMap[aggregation || QueryViewerAggregationType.Sum](
+) => {
+  return aggregateMap[aggregation || QueryViewerAggregationType.Sum](
     values,
     quantities
   );
+};
 
 function aggregateDatum(
```

Comment:
`[aggregateDatum]` It would be nice to add some tests using this utility with different values to validates the new behavior.